pilot/pkg/model/context.go

```diff
@@ -341,6 +341,10 @@ type WatchedResource struct {
 	// For Delta Xds, all resources of the TypeUrl that a client has subscribed to.
 	ResourceNames []string
 
+	// CacheKeys maps resource names to actual cache entries, which contain the dependencies
+	// of that resource. We use this for delta.
+	CacheKeys map[string]XdsCacheEntry
```

Comment:
I think this is very mem costing, can you have a bench mark. If it is the same scalability as @ramaraochavali's recent scenario. I can not imagine the mem usage.

Reply:
what was @ramaraochavali's recent scenario?

and what would we be benchmarking here? as number of clusters increases, the size of the map will increase right? do we benchmark for the size of `XdsCacheEntry`?